pkg/controller/velero/velero.go

```diff
@@ -135,7 +145,11 @@ func (r *ReconcileVelero) provisionVelero(reqLogger logr.Logger, namespace strin
 	default:
 		return reconcile.Result{}, fmt.Errorf("unable to determine platform")
 	}
-	if err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: credentialsRequestName}, foundCr); err != nil {
+	crName := types.NamespacedName{
+		Namespace: cr.ObjectMeta.Namespace,
+		Name:      cr.ObjectMeta.Name,
+	}
+	if err = r.client.Get(context.TODO(), crName, foundCr); err != nil {
```

Comment:
Oh, I like this. Let's try something even a little but more clever, and do it for all the objects:
```suggestion
	crName, err := runtimeClient.ObjectKeyFromObject(cr)
	if err != nil {
		return reconcile.Result{}, err
	}
	if err = r.client.Get(context.TODO(), crName, foundCr); err != nil {
```

Reply:
Ah ha!  Did not know about that function.  :+1:

Reply:
Stole this \o/